.github/SwiftCurrentLint/SwiftCurrent-IRGeneratorTests/SwiftCurrent_IRGeneratorTests.swift

```diff
@@ -0,0 +1,235 @@
+//
+//  SwiftCurrent_IRGeneratorTests.swift
+//  SwiftCurrent
+//
+//  Created by Tyler Thompson on 3/3/22.
+//  Copyright © 2022 WWT and Tyler Thompson. All rights reserved.
+//  
+
+import Foundation
+import ShellOut
+import XCTest
+
+class SwiftCurrent_IRGeneratorTests: XCTestCase {
+    static var packageSwiftDirectory: URL = {
+        // ../../../ brings you to SwiftCurrent directory
+        URL(fileURLWithPath: #file).deletingLastPathComponent().deletingLastPathComponent().deletingLastPathComponent().deletingLastPathComponent()
+    }()
+
+    lazy var generatorCommand: String = {
+        "\(Self.packageSwiftDirectory.path)/.build/*/debug/SwiftCurrent_IRGenerator"
+    }()
+
+    override class func setUp() {
+        XCTAssertNoThrow(try shellOut(to: "rm -rf \(Self.packageSwiftDirectory.path)/.build/*/debug"))
+        XCTAssert(try shellOut(to: "cd \(Self.packageSwiftDirectory.path) && swift build --product=SwiftCurrent_IRGenerator").contains("Build complete!"))
+    }
+
+    func testSingleDecodableStruct() throws {
+        let source = """
+        struct Foo: WorkflowDecodable { }
+        """
+
+        let output = try shellOut(to: "\(generatorCommand) \"\(source)\"")
+        let IR = try JSONDecoder().decode([IRType].self, from: XCTUnwrap(output.data(using: .utf8)))
+
+        XCTAssertEqual(IR.count, 1)
+        XCTAssertEqual(IR.first?.name, "Foo")
+    }
+
+    func testSingleDecodableClass() throws {
+        let source = """
+        class Foo: WorkflowDecodable { }
+        """
+
+        let output = try shellOut(to: "\(generatorCommand) \"\(source)\"")
+        let IR = try JSONDecoder().decode([IRType].self, from: XCTUnwrap(output.data(using: .utf8)))
+
+        XCTAssertEqual(IR.count, 1)
+        XCTAssertEqual(IR.first?.name, "Foo")
+    }
+
+    func testMultipleMixedStructsAndClasses() throws {
+        let source = """
+        class Foo: WorkflowDecodable { }
+        struct Bar: WorkflowDecodable { }
+        """
+
+        let output = try shellOut(to: "\(generatorCommand) \"\(source)\"")
+        let IR = try JSONDecoder().decode([IRType].self, from: XCTUnwrap(output.data(using: .utf8)))
+            .sorted { $0.name < $1.name }
+
+        XCTAssertEqual(IR.count, 2)
+        XCTAssertEqual(IR.first?.name, "Bar")
```

Comment:
using .first and .last -- Is order guaranteed/significant/important/non-changeable? 

Consider IR.map(\.name).contains("Bar") and IR.map(\.name).contains("Foo")

Reply:
Check out line 60, that's why it's sorted, for guaranteed order.

using `contains` does not produce as friendly of an error message upon failing. I guess I could write my own message, but this way is lazier